lib/background_worker.rb

```diff
@@ -1,30 +1,47 @@
-module BackgroundWorker
-  VERSION = '0.0.1'
-
-  require 'background_worker/base'
-  require 'background_worker/persistent_state'
-
-  def self.after_exception(&block)
-    if block
-      @@after_exception = block
-    else
-      @@after_exception if defined?(@@after_exception)
-    end
-  end
+require 'background_worker/version'
+require 'background_worker/uid'
+require 'background_worker/base'
+require 'background_worker/persistent_state'
 
-  # Provide your own background worker enqueue implementation
-  #   eg:
-  #  Resque.enqueue(klass, method_name, options)
+module BackgroundWorker
   def self.enqueue(klass, method_name, options)
-    if @@enqueue_with
-      @@enqueue_with.call(klass, method_name, options)
+    if enqueue_with
+      enqueue_with.call(klass, method_name, options)
     else
       # No backgrounded by default
       klass.perform(method_name, options)
     end
   end
 
-  def self.enqueue_with(&block)
-    @@enqueue_with = block
+  class << self
+    # Provide a logger
+    attr_writer :logger
+
+    attr_writer :after_exception
+
+    # Provide your own background worker enqueue implementation
+    #
+    # eg:
+    # BackgroundWorker.enqueue_with = -> klass, method_name, options {
+    #   Resque.enqueue(klass, method_name, options)
+    # }
+    attr_writer :enqueue_with
+  end
+
+  def self.logger
+    @logger ||= Logger.new(STDOUT)
+  end
+
+  def self.verify_active_connections!
+    Rails.cache.reconnect if defined?(Rails)
+    ActiveRecord::Base.verify_active_connections! if defined?(ActiveRecord)
+  end
+
+  def self.after_exception(&_block)
+    @after_exception ||= lambda  do |e|
+      logger.error '** No after_exception handler installed **'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.